### PR TITLE
GODRIVER-2031 Document that Cursor isn't goroutine safe

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

File: mongo/cursor.go
Comment (kevinAlbs) [inline mongo/cursor.go:25]: Nice, I see this warning on the `SessionContext` type too.

DRIVERS-1789 also notes change stream cursors, since a user may be blocked on the change stream's `Next` call indefinitely. To unblock, users can cancel the passed context. But I think the motivation for the documentation is to make it clear that you cannot call `Close` in a separate goroutine in that case.

Can you add the same note to the `ChangeStream` type?

Comment (iwysiu) [reply mongo/cursor.go:25]: done

**Review by kevinAlbs — APPROVED** [review]

**Review by matthewdale — APPROVED** [review]
👍

**Pull request merged**
